databroker/_drivers/jsonl.py

```diff
@@ -102,6 +102,10 @@ def __init__(self, paths, *, handler_registry=None, root_map=None,
             same methods as ``DocumentRouter``.
         query : dict, optional
             Mongo query that filters entries' RunStart documents
+        transforms : dict
```

Comment:
I think this description could be slightly more clear. Let's workshop it a bit. cc @ronpandolfi @dylanmcreynolds @prjemian @mrakitin 

My pitch:

> A dict that maps any subset of the keys {``start``, ``stop``, ``resource``, ``descriptor``} to a function that accepts a document of the corresponding type and returns it, potentially modified. This feature is for patching up erroneous metadata. It is intended for quick, temporary fixes that may later be applied permanently to the data at rest (e.g via a database migration).

Too verbose? I want to give a clear sense of the intended scope of this feature, which could be at risk of being abused.

As a reminder, the immediate motivation for this feature is that we encountered Event Descriptors that recorded the wrong array shape, breaking code that needs to know the correct shape in advance. Ultimately, we want to fix the code that is reporting the incorrect shape and then go back and do a database migration to update the existing documents. But in the short term, and to test the fix, we want this hook to let us patch up the documents on their way out of databroker.

Reply:
updated with your suggestion

Reply:
Yes, this will be very useful for testing migrations. The description is clear; better type hinting would help. `Dict[str, Callable]`?
Why preclude patching of event and datums?

Reply:
Good question. We're excluding Event and Datum for now because:
- Adding the overhead of a function call (~200 ms) to every row would be a much more significant performance overhead. As is, this overhead will be imposed ~10 times for a typical Run and no more than ~100 times in extreme cases. If we were to include Event and Datum that would grow to ~10^7 at some beamlines we support.
- Making it possible to use this to edit _data_ (i.e. changing the values in Events) feels like a big step up in scope at best and maybe something we flat-out shouldn't support: one could argue that the possibility of a badly-configured transformer is changing your data under you  is would be an anti-feature. An alternative: copy the data into a new Run (with a new unique identifier).
- We have rarely (never?) needed to patch up Event or Datum documents in the wild. The only pressing need is to fix up Event Descriptors.